lib/spack/spack/install_test.py

```diff
@@ -52,6 +93,638 @@ def get_test_stage_dir():
     )
 
 
+def cache_extra_test_sources(pkg: "spack.package_base.PackageBase", srcs: ListOrStringType):
+    """Copy relative source paths to the corresponding install test subdir
```

Comment:
Is it worth to check upfront that the paths are relative? Or is this is this function usually called in a context where it wouldn't add much value?

Reply:
This is called by users so that's an excellent point about checking the precondition.

Reply:
Done.